File: serviciu_bilete/src/main/java/com/pos/config/OpenApiConfig.java
package com.pos.config;

import io.swagger.v3.oas.models.Components;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.security.SecurityScheme;
import io.swagger.v3.oas.models.servers.Server;
import org.springdoc.core.models.GroupedOpenApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.List;

@Configuration
public class OpenApiConfig {

    @Bean
    public OpenAPI eventFlowOpenAPI() {
        final String securitySchemeName = "bearerAuth";
        return new OpenAPI()
                .info(new Info()
                        .title("EventFlow API")
                        .description("Event Ticketing Management System - RESTful API")
                        .version("1.0.0")
                        .contact(new Contact()
                                .name("Robert")
                                .email("robert-constantin.grigoras@student.tuiasi.ro")))
                .servers(List.of(
                        new Server()
                                .url("http://localhost:8080")
                                .description("Development Server")
                ))
                .components(new Components()
                        .addSecuritySchemes(securitySchemeName,
                                new SecurityScheme()
                                        .name(securitySchemeName)
                                        .type(SecurityScheme.Type.HTTP)
                                        .scheme("bearer")
                                        .bearerFormat("JWT")
                                        .description("Enter JWT token from IDM service")));
    }

    @Bean
    public GroupedOpenApi publicApi() {
        return GroupedOpenApi.builder()
                .group("eventflow-api")
                .packagesToScan("com.pos.controller")
                .pathsToMatch("/api/**")
                .build();
    }
}

File: serviciu_clienti/src/main/kotlin/com/pos/serviciu_clienti/config/OpenApiConfig.kt
package com.pos.serviciu_clienti.config


import io.swagger.v3.oas.models.Components
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.security.SecurityScheme
import io.swagger.v3.oas.models.servers.Server
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfig {

    @Bean
    fun clientServiceOpenAPI(): OpenAPI {
        val securitySchemeName = "bearerAuth"
        return OpenAPI()
            .info(
                Info()
                    .title("EventFlow - Client Service API")
                    .description("Serviciu RESTful pentru gestiunea clienților si biletelor achiziționate. Parte din platforma EventFlow pentru gestiunea evenimentelor.")
                    .version("1.0.0")
                    .contact(
                        Contact()
                            .name("Robert")
                            .email("robert-constantin.grigoras@student.tuiasi.ro")
                    )
            )
            .servers(
                listOf(
                    Server()
                        .url("http://localhost:8081")
                        .description("Development Server - Client Service")
                )
            )
            .components(
                Components()
                    .addSecuritySchemes(
                        securitySchemeName,
                        SecurityScheme()
                            .name(securitySchemeName)
                            .type(SecurityScheme.Type.HTTP)
                            .scheme("bearer")
                            .bearerFormat("JWT")
                            .description("Enter JWT token from IDM service")
                    )
            )
    }
}